src/main_thread/decrypt/find_key_system.ts

```diff
@@ -162,13 +162,24 @@ function buildKeySystemConfigurations(
   keySystemTypeInfo: IKeySystemType,
 ): MediaKeySystemConfiguration[] {
   const { keyName, keyType, keySystemOptions: keySystem } = keySystemTypeInfo;
-  const sessionTypes = ["temporary"];
+  let sessionTypes: string[];
   let persistentState: MediaKeysRequirement = "optional";
   let distinctiveIdentifier: MediaKeysRequirement = "optional";
 
-  if (!isNullOrUndefined(keySystem.persistentLicenseConfig)) {
+  if (Array.isArray(keySystem.wantedSessionTypes)) {
+    sessionTypes = keySystem.wantedSessionTypes;
+    if (
+      arrayIncludes(keySystem.wantedSessionTypes, "persistent-license") &&
+      !isNullOrUndefined(keySystem.persistentLicenseConfig)
+    ) {
+      persistentState = "required";
+    }
+  } else if (!isNullOrUndefined(keySystem.persistentLicenseConfig)) {
     persistentState = "required";
-    sessionTypes.push("persistent-license");
+    // TODO: shouldn't it be just `["persistent-license"]` here?
+    sessionTypes = ["temporary", "persistent-license"];
```

Comment:
Can this TODO be resolved ?

Reply:
I can but I was afraid of breaking things.

Though theoretically, it should resolve cases here, not fail, I'll do a supplementary commit.

Reply:
done

Reply:
Some unit tests are failing

Reply:
Fixed, it was linked to the old behavior